Source/Diagnostics/ReducedDiags/ParticleEnergy.cpp

```diff
@@ -0,0 +1,146 @@
+#include "ParticleEnergy.H"
+#include "WarpX.H"
+#include "WarpXConst.H"
+#include "AMReX_REAL.H"
+#include "AMReX_ParticleReduce.H"
+#include <iostream>
+#include <cmath>
+
+using namespace amrex;
+
+/// constructor
+ParticleEnergy::ParticleEnergy (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    /// get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    /// get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    /// get number of species (int)
+    auto species_number = mypc.nSpecies();
+
+    /// get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            /// open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + ".txt",
+                std::ofstream::out | std::ofstream::app);
+            /// write header row
+            ofs << "#";
+            ofs << "step";
+            ofs << m_sep;
+            ofs << "time(s)";
+            ofs << m_sep;
+            ofs << "total(J)";
+            for (int i = 0; i < species_number; ++i)
+            {
+                ofs << m_sep;
+                ofs << species_names[i]+"(J)";
+            }
+            ofs << m_sep;
+            ofs << "total.mean(J)";
+            for (int i = 0; i < species_number; ++i)
+            {
+                ofs << m_sep;
+                ofs << species_names[i]+".mean(J)";
+            }
+            ofs << std::endl;
+            /// close file
+            ofs.close();
+        }
+    }
+
+}
+///< end constructor
+
+/// function that computes kinetic energy
+void ParticleEnergy::ComputeDiags (int step)
+{
+
+    /// Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    /// get MultiParticleContainer class object
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+
+    /// get number of species (int)
+    auto species_number = mypc.nSpecies();
```

Comment:
`species_number` is a bit confusing, could we rename this variable `nSpecies` or `number_of_species` or something more explicit?